huff_utils/src/abi.rs

```diff
@@ -259,6 +257,17 @@ impl FunctionParamType {
     /// Convert string to type
     pub fn convert_string_to_type(string: &str) -> Self {
         let input = string.to_string().to_lowercase();
+        let split_input: Vec<&str> = input.split('[').collect();
+        if split_input.len() > 1 {
+            let mut cleaned: Vec<String> = split_input
+                .iter()
+                .map(|x| x.replace(']', ""))
+                .map(|x| if x.is_empty() { "0".to_owned() } else { x })
+                .collect();
+            let func_type = FunctionParamType::convert_string_to_type(&cleaned.remove(0));
+            let sizes: Vec<usize> = cleaned.iter().map(|x| x.parse::<usize>().unwrap()).collect();
```

Comment:
Can we add a tracing error here - or preferably return a `Result<Self, ParserError>` on bad usize coalescing?

Reply:
ibid below in the function on usize unwraps - realistically won't ever happen but cleaner error bubbling itf